test/e2e/utils/observer.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2023 ScyllaDB.
+
+package utils
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/scylladb/scylla-operator/pkg/kubeinterfaces"
+	"k8s.io/apimachinery/pkg/util/wait"
+	watchutils "k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/watch"
+)
+
+type ObserverEvent[T kubeinterfaces.ObjectInterface] struct {
+	Action watchutils.EventType
+	Obj    T
+}
+
+type ObjectObserver[T kubeinterfaces.ObjectInterface] struct {
+	Events []ObserverEvent[T]
+
+	lw cache.ListerWatcher
+
+	resultChan chan error
```

Comment:
The name is misleading for a channel dedicated to errors

Reply:
done